### PR TITLE
DEV-26980 Improve error response

### PR DESCRIPTION
**Pull request by mrnyax**
Update docs with more friendly error message when creating or editing user then assigning more privileges than the current user

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -2118,6 +2118,19 @@ This is an alternative way of getting information about the current (in request
             }
         }
 
++ Response 403 (application/json)
+
+        // when you can't assign privileges you don't have when you create users
+        {
+        "links": {},
+            "error": {
+                "detail": "The user does not have the required privileges to perform the operation. It’s not possible to make these changes. You can only create or edit a user with privileges that also apply to your user account.",
```
Comment (madeline-brennan) [inline apiary.apib:2127]:
```suggestion
                "detail": "It’s not possible to make these changes. You can only create or edit a user with privileges that also apply to your user account.",
```

File: apiary.apib [hunk 2]
```diff
@@ -3203,6 +3216,19 @@ This is an alternative way of getting information about the current (in request
             }
         }
 
++ Response 403 (application/json)
+
+        // when you can't assign privileges you don't have when you edit users
+        {
+        "links": {},
+            "error": {
+                "detail": "The user does not have the required privileges to perform the operation. It’s not possible to make these changes. You can only create or edit a user with privileges that also apply to your user account.",
```
Comment (madeline-brennan) [inline apiary.apib:3225]:
```suggestion
                "detail": "It’s not possible to make these changes. You can only create or edit a user with privileges that also apply to your user account.",
```

**Review by george-haddad — APPROVED** [review]
Looks fine to me with Madeline's suggestions

**Review by tkrisztian95 — APPROVED** [review]
Looks good!

**Pull request merged**